contracts/governance/test/ZeroExVotesTest.t.sol

```diff
@@ -53,6 +55,53 @@ contract ZeroExVotesTest is BaseTest {
         votes.initialize(account2, quadraticThreshold);
     }
 
+    function testShouldBeAbleToMigrate() public {
+        vm.roll(block.number + 1);
+
+        vm.startPrank(account2);
+        token.approve(address(wToken), 100e18);
+        wToken.depositFor(account2, 100e18);
+        wToken.delegate(account3);
+        vm.stopPrank();
+
+        vm.startPrank(account3);
+        token.approve(address(wToken), 200e18);
+        wToken.depositFor(account3, 200e18);
+        wToken.delegate(account3);
+        vm.stopPrank();
+
+        vm.roll(block.number + 1);
+
+        ZeroExVotesMigration newImpl = new ZeroExVotesMigration();
+        assertFalse(address(uint160(uint256(vm.load(address(votes), 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc)))) ==  address(newImpl));
+        vm.prank(account1);
+        votes.upgradeToAndCall(address(newImpl), abi.encodeWithSignature("initialize()"));
+        assertEq(address(uint160(uint256(vm.load(address(votes), 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc)))), address(newImpl));
+
+        vm.roll(block.number + 1);
+
+        vm.prank(account2);
+        wToken.transfer(address(this), 50e18);
+        vm.prank(account3);
+        wToken.transfer(address(this), 100e18);
+    }
```

Comment:
Observe that we can't migrate to new forms of voting tabulation because balances need to be tagged with the blockheight at which they were last updated. The lined in `ZeroExVotesMigration`  with the comment `// underflow` is the culprit.

Reply:
Added `DelegateInfo.balanceLastUpdated` block number stamps to the wrapped token.